arch/risc-v/Kconfig

```diff
@@ -461,19 +461,51 @@ config ARCH_RV_MMIO_BITS
 	default 32 if ARCH_RV32
 	default 64 if ARCH_RV64
 
+config ARCH_RV_CPUID_MAP
+	bool "Enable CPUID mapping"
+	default n
+	---help---
+		Enable CPUID mapping for systems where the hardware CPU IDs
+		need to be mapped to logical CPU IDs. This is useful for
+		systems with non-contiguous or non-linear CPU numbering.
+
 config ARCH_RV_HARTID_BASE
 	int "Base hartid of this cluster"
+	depends on ARCH_RV_CPUID_MAP
```

Comment:
On second thought, we need to remove this depends on, it will cause a regression and there is really no reason for adding this depends on. Sorry for wasting time...